src/app/sevices/taskFormValidation.service.ts

```diff
@@ -1,16 +1,23 @@
 import { Injectable } from '@angular/core';
 import { TaskService } from './task.service';
 import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { AppState } from '../store/app.store';
+import { Store } from '@ngrx/store';
+import { Task } from '../models/task.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TaskFormValidationService {
-  constructor(private taskService: TaskService) {}
+  constructor(private taskService: TaskService, private store: Store<AppState>) {}
 
   uniqueTitle(title?: string): ValidatorFn {
     return (control: AbstractControl): ValidationErrors | null => {
-      const task = this.taskService.getTaskByTitle(control.value);
+      let tasks: Task[] = []
+      this.store.select(state => state.task.tasks).subscribe(storeTasks => {
```

Comment:
Everytime, when function inside the uniqueTitle validator is called, the new subscription is created. And they are never removed. This is causing a memory leak.